### PR TITLE
Bug 1142680 - (cleanup) Write a compare-talos equivalent and integrate i...

### PR DESCRIPTION
**Pull request by jmaher**
...t into PerfHerder. r=wlach

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder-ui/478)

### CODE REVIEW

**Comment by edmorley** [comment]
This can wait until wlach has given feedback, since you can do it at the same time, but could you update the commit message from "Bug 1142680 - (cleanup) Write a compare-talos equivalent and integrate it into PerfHerder" - to eg "Bug 1142680 - Clean up compareperf" or something clearer? :-)

**Comment by jmaher** [comment]
how do you change the commit message?

**Comment by edmorley** [comment]
https://ateam-bootcamp.readthedocs.org/en/latest/reference/git_github.html#rebasing-commits

File: webapp/app/js/compareperf.js [hunk 1]
```diff
@@ -104,7 +105,7 @@ comparePerf.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$ro
                                                  $stateParams.e10s);
 
             if (seriesSummary != null && seriesSummary.signature !== undefined) {
-              seriesList.push(seriesSummary);
+              rawResultsMap[seriesSummary.signature] = {name: seriesSummary.name, platform: seriesSummary.platform};
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: Could you break out the dictionary onto a new line like:
rawResultsMap[seriesSummary.signature] = {
   name: seriesSummary.name,
   platform: seriesSummary.platform
};

File: webapp/app/js/compareperf.js [hunk 2]
```diff
@@ -155,7 +152,7 @@ comparePerf.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$ro
                                                      $stateParams.e10s);
 
                 if (seriesSummary != null && seriesSummary.signature !== undefined) {
-                  newSeriesList.push(seriesSummary);
+                  newRawResultsMap[seriesSummary.signature] = {name: seriesSummary.name, platform: seriesSummary.platform};
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: Likewise, this would be easier to read if broken out to multiple lines.

File: webapp/app/js/compareperf.js [hunk 3]
```diff
@@ -214,25 +212,14 @@ comparePerf.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$ro
...
-        compareResultsMap[counter++] = {'name': testName.replace(' summary', ''),
-                                        'isEmpty': true, 'isMinor': false, 'headerColumns': 2,
-                                        'originalGeoMean': 'Old Rev', 'originalStddev': 'StdDev',
-                                        'newGeoMean': 'New Rev', 'newStddev': 'StdDev',
-                                        'delta': 'Delta', 'deltaPercentage': 'Delta'};
-
+        $scope.compareResults[testName] = [];
 
         $scope.platformList.forEach(function(platform) {
           var cmap = {'originalGeoMean': NaN, 'originalRuns': 0, 'originalStddev': NaN,
```
Comment (wlach) [inline webapp/app/js/compareperf.js]: While you're here, could you remove the quotes around the various properties? Would make it easier to read.

**Comment by wlach** [comment]
This looks good! Just some minor comments. And yeah, please update the commit message per @edmorley's suggestion. You can do that, as well as avoid creating new commits, by running `git commit --amend --reset-author` instead of `git commit`. Just use git push -f when pushing to your branch (to avoid complaints about overwriting stuff).

**Comment by jmaher** [comment]
updated with all feedback addressed including the commit message.  Thanks guys.  I will say if we are going to point folks at the bootcamp (https://ateam-bootcamp.readthedocs.org/en/latest/reference/git_github.html#rebasing-commits) for git commands, etc. we should really invest some time into making that more complete.

**Comment by jmaher** [comment]
this got lost, added in https://github.com/mozilla/treeherder-ui/pull/485

**Pull request closed without merging**